### ISSUE TITLE
[date-picker]: once a range is selected you can pick an earlier start date but not later

### ISSUE THREAD

**Issue by Elijbet** [Esri/calcite-components#6223]
### Actual Behavior

Currently, the `date-picker` does not allow for the user to reset the `start-date` to a later date. 

https://user-images.githubusercontent.com/19231036/210434893-8d640754-d384-4e8e-9edc-83d2a7068a26.mov

Looking for feedback on whether the suggested behavior is acceptable. 

@macandcheese @SkyeSeitz and anyone else who has input.

### Expected Behavior

Suggested expected behavior can be seen in this example:

https://vcalendar.io/datepicker.html

The third click after the start and end date is selected should start it over. 

### Reproduction Sample

Video attached. 

### Reproduction Steps

Video attached. 

### Reproduction Version

v1.0.0-beta.99

### Esri team

Calcite (dev)

**Comment by SkyeSeitz** [comment]
I agree with the proposed expected behavior as seen in the example referenced: 

https://user-images.githubusercontent.com/29716057/210437257-7ea29381-9483-4410-b619-f697a693d4f9.mov

**Comment by macandcheese** [comment]
I think there are some other issues with this component - the `proximity-selection-disabled` property is supposed to allow the first date to be re-chosen, but there are clearly some issues with this, as shown: 


https://user-images.githubusercontent.com/4733155/210440079-44804905-9965-4bb5-b914-475585baa2cb.mov

**Comment by benelan** [comment]
Note - linking to storybook or the doc is a valid repro sample for the bot, e.g.
- https://esri.github.io/calcite-components/?path=/story/components-controls-datepicker--range
- https://developers.arcgis.com/calcite-design-system/components/date-picker/#sample

**Comment by benelan** [comment]
> I think there are some other issues with this component - the proximity-selection-disabled property is supposed to allow the first date to be re-chosen, but there are clearly some issues with this

Yeah the component is pretty messed up right now. Here are some additional issues:
#6198
#6039
#5886
#5969
#5816
#6194

I think a lot of those are recent regressions. I pinned down the breaking commit for the ones I logged, but I can use `git bisect` to do the rest of them too so we can figure out what's causing all of this stuff. So far it seems like the regressions are coming from  #5459 and the [deprecation removals](https://github.com/Esri/calcite-components/pulls?q=is%3Apr+is%3Aclosed+author%3Aeriklharper+in%3Atitle+date-picker+merged%3A2022-12-01..2022-12-22). The components completely changed in those PRs so it seems likely some of the other regressions originated there as well.

I'm not sure if we will have time to fix all of this before v1, so we may need to revert some PRs if possible. We can't revert the deprecation removals but maybe #5459?

**Comment by Elijbet** [comment]
To recap the last meeting on this issue: 

For now, let's see if we can fix `proximity-selection-disabled`, whether it's behavior or styling, and revisit the 3rd-click approach post v1, as resetting on the 3rd click may not be ideal all of the time (I'll let @jcfranco elaborate on this a bit more for design to consider).

If Design can confirm that the 3rd click UX would get rid of `proximity-selection-disabled` and that it's straightforward to implement, we can consider it post v1. There's been feedback about the current UX being very confusing.

**Comment by jcfranco** [comment]
@Elijbet Thanks for posting that here. Regarding my 3rd-click not being ideal scenario, here are the steps I think could be frustrating for users:

1. User clicks on March 1st as start date
2. User clicks on March 6th as end date, range is created
3. User wants to extend the range to end on March 7th with a 3rd click

**With the current behavior**
4a. The end date is updated to March 7th and the range is updated accordingly

**With the 3rd-click approach**
4b. The range is reset and March 7th is the first date in the range
5b. User clicks on March 1st as the start date and the range is updated accordingly

**TL;DR**: A 3rd-click approach to reset introduces one more step for users to update ranges

**Comment by ashetland** [comment]
> There's been feedback about the current UX being very confusing...

Do we have specifics on this? Is it because `date-picker` is being used in a context without an input?

**Comment by geospatialem** [comment]
An estimate will need some additional context on the implementation decided.

**Comment by paulcpederson** [comment]
> Do we have specifics on this? Is it because date-picker is being used in a context without an input?

We have at least one of these and yes, when there is no input (which I think is valid) it's impossible to make the start day later after you've selected a range. 

> When `true`, disables the default behavior on the third click of narrowing or extending the range and instead starts a new range.

We would opt into `proximity-selection-disabled` but it doesn't work. Can we get `proximity-selection-disabled` fixed? Right now the docs are incorrect and the component when using this prop is really just kind of broken. Not sure why this was moved to the freezer.

**Comment by ashetland** [comment]
Thanks @paulcpederson for resurfacing this. Agree we should fix `proximity-selection-disabled` to act as described, "This disables the default behavior on the third click of narrowing or extending the range and instead starts a new range."

When used separate from an input controlling the start and end, a 3rd click resetting the range feels appropriate.

**Comment by DitwanP** [comment]
🍠 Verified on `2.8.0-next.7`
